### PR TITLE
Initial version of contributing guide.

### PR DESCRIPTION
**Pull request by ludgerpaehler**
Addition of an initial version of a contribution guide based on the LLVM contribution guide, and pointing to the LLVM code of conduct.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+## Ways to Contribute
+
+### Bug Reports and Feature Requests
+
+If you are working with Enzyme, and run into a bug, or require additional features, we definitely want to know about it. Please let us know by either creating a [GitHub Issues](https://github.com/EnzymeAD/Enzyme/issues), or starting a thread in Enzyme's [LLVM Discourse Category](https://discourse.llvm.org/c/projects-that-want-to-become-official-llvm-projects/enzyme/45).
+
+### Bug Fixes
+
+If you are interested in contributing code to Enzyme, issues labeled with the [good-first-issue](https://github.com/EnzymeAD/Enzyme/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) keyword in the [GitHub Issues](https://github.com/EnzymeAD/Enzyme/issues) are a good way to get familiar with the code base. If you are interested in fixing a bug please comment on it to let people know you are working on it.
+
```
Comment (wsmoses) [inline CONTRIBUTING.md:15]: I would actually recommend trying it out in the enzyme compiler explorer here.

Comment (ludgerpaehler) [reply CONTRIBUTING.md:15]: I just added a specific highlight to the compiler explorer.

Comment (wsmoses) [reply CONTRIBUTING.md:15]: Oh I meant as a first way to test a bug, rather than compiling LLVM from source.

Comment (ludgerpaehler) [reply CONTRIBUTING.md:15]: How about now, i.e. Compiler Gym as the first default option, and only then resorting to a source build..

**Review by wsmoses — APPROVED** [review]

**Review by vchuravy — APPROVED** [review]

**Pull request merged**